src/AppInstallerCLICore/Workflows/ConfigurationFlow.cpp

```diff
@@ -187,6 +188,39 @@ namespace AppInstaller::CLI::Workflow
                 {
                     factoryMap.Insert(ConfigurationRemoting::ToHString(ConfigurationRemoting::PropertyName::DscExecutablePath), Utility::ConvertToUTF16(context.Args.GetArg(Args::Type::ConfigurationProcessorPath)));
                 }
+                else
+                {
+                    // Make sure DSC executable path can be found. Otherwise, we'll install the DSC v3 package.
+                    winrt::hstring foundExecutablePath = factoryMap.Lookup(ConfigurationRemoting::ToHString(ConfigurationRemoting::PropertyName::FoundDscExecutablePath));
+                    if (foundExecutablePath.empty())
+                    {
+                        AICLI_LOG(Config, Info, << "dsc.exe not found and not provided. Installing dsc package from store.");
+
+                        auto installDscContextPtr = context.CreateSubContext();
+                        Execution::Context& installDscContext = *installDscContextPtr;
+                        auto previousThreadGlobals = installDscContext.SetForCurrentThread();
+
+                        Manifest::ManifestInstaller dscInstaller;
+// TEMP: Until DSCv3 support is not experimental, allow the preview build to be installed automatically.
+// #ifndef AICLI_DISABLE_TEST_HOOKS
+                        dscInstaller.ProductId = "9PCX3HX4HZ0Z";
+// #else
+//                      dscInstaller.ProductId = "9NVTPZWRC6KQ";
+// #endif
+                        installDscContext.Add<Execution::Data::Installer>(std::move(dscInstaller));
+                        installDscContext.Args.AddArg(Execution::Args::Type::InstallScope, Manifest::ScopeToString(Manifest::ScopeEnum::User));
+                        installDscContext.Args.AddArg(Execution::Args::Type::Silent);
+                        installDscContext.Args.AddArg(Execution::Args::Type::Force);
+
+                        installDscContext << MSStoreInstall;
+
+                        if (installDscContext.IsTerminated())
+                        {
+                            AICLI_LOG(Config, Error, << "Could not find dsc.exe, it must be provided by the user.");
+                            THROW_WIN32(ERROR_FILE_NOT_FOUND);

```

Comment:
User message and use `TERMINATE_CONTEXT` instead of throwing directly.

Reply:
The MSStoreInstall workflow task prints installation failed message with error code to the screen. I guess we don't need a duplicate install failed message here.

Reply:
I'm more concerned about the mitigation message than the "install failed".  Inform the user that they can provide the argument for the location of dsc.